src/software/ai/passing/pass_generator_test.cpp

```diff
@@ -61,14 +56,11 @@ class PassGeneratorTest : public testing::Test
             // time has expired, whichever comes first. We also check that the score
             // is not small, otherwise we can get "false convergence" as the
             // pass just starts to "move" towards the converged point
-        } while ((std::abs(curr_score - prev_score) > max_score_diff ||
-                  (std::abs(curr_score - prev_prev_score) > max_score_diff) ||
-                  curr_score < 0.01) &&
-                 seconds_so_far < max_num_seconds);
-
-        ASSERT_LT(seconds_so_far, max_num_seconds)
-            << "Pass generator did not converge after running for " << max_num_seconds
-            << " seconds";
+            if (std::abs(curr_score - prev_score) < min_score_diff)
+            {
+                break;
+            }
+        }
```

Comment:
Do we not need an `ASSERT` in this function anymore?

Reply:
We don't, that was for some checks I had that were needed when things weren't deterministic. Looking at this made me realize that there is a check we could use here though, added:
```
        EXPECT_LE(std::abs(curr_score - prev_score), min_score_diff);
```
after the loop.